pkg/snapshot/layered_map.go

```diff
@@ -17,20 +17,27 @@ limitations under the License.
 package snapshot
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
 	"path/filepath"
 	"strings"
+
+	"github.com/GoogleContainerTools/kaniko/pkg/util"
 )
 
 type LayeredMap struct {
-	layers    []map[string]string
-	whiteouts []map[string]string
-	hasher    func(string) (string, error)
+	layers      []map[string]string
+	whiteouts   []map[string]string
+	added       []map[string]string
+	hasher      func(string) (string, error)
+	cacheHasher func(string) (string, error)
```

Comment:
What is the difference between hasher and cacheHasher?

Reply:
nvm, read the description again. Can you add comments here describing hasher and cacheHasher?

Reply:
cacheHasher doesn't include file mtime in the hash, since that would be different for every build and we need a stable key for caching

Reply:
added the comment, there's also more detailed descriptions in util.go where the hash functions live